x/market/types/message_deal.go

```diff
@@ -71,3 +71,44 @@ func (msg *MsgCreateDeal) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{creator}
 }
+
+var _ sdk.Msg = &MsgSellData{}
+
+func NewSellData(cert DataValidationCertificate, seller string) *MsgSellData {
+	return &MsgSellData{
+		Cert:   &cert,
+		Seller: seller,
+	}
+}
+
+func (msg *MsgSellData) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgSellData) Type() string {
+	return "SellData"
+}
+
+// ValidateBasic is validation for MsgSellData.
+// TODO: Basic Validation for MsgSellData.
+func (msg *MsgSellData) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Seller)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+
+	return nil
+}
```

Comment:
Shouldn't we check here if the seller and the requester (in data certificate) are the same?

Reply:
Yes I didn't fix yet. I will fix and push it soon!